harness/determined/deploy/aws/templates/simple-rds.yaml

```diff
@@ -469,9 +474,13 @@ Resources:
     DeletionPolicy: Delete
     Properties:
       DBInstanceClass: !Ref DBInstanceType
+      DBSnapshotIdentifier:
+        !If [NoDBSnapshot, !Ref "AWS::NoValue", !Ref DBSnapshot]
       Engine: postgres
-      MasterUsername: postgres
-      MasterUserPassword: !Ref DBPassword
+      MasterUsername:
+        !If [NoDBSnapshot, "postgres", !Ref "AWS::NoValue"]
+      MasterUserPassword:
+        !If [NoDBSnapshot, !Ref DBPassword, !Ref "AWS::NoValue"]
```

Comment:
Will the user need to pass something here? Otherwise there will be no value in the master config and it would fail to connect?

Reply:
In my testing, this just caused the DB to generate a password on setup.
That's AWS behavior, so I'm not sure if that's the intended behavior for `det deploy`, but it's pretty sensible.

Reply:
but how would the det master know the generated password?

Reply:
yeah, ^ my thoughts exactly

Reply:
So I see how the problem would manifest if we were providing the template to a user to deploy manually, but I'm not sure how that affects the setup here.
The password flag isn't currently required since it's set to some default, and things deploy using that default currently. This portion of the template doesn't affect if/how the password is rendered into any `master.yml` config file -- is a password is passed, it'll still be in the master config generated, if it isn't then the master config is still generated with the same default password.

I'm not sure I understand if the question is about what I've implemented or about `det deploy`'s behavior in general rn

Reply:
I think we should just remove this and configure the database with a password still. I've tested it, it works fine and it's much generally useful. It's much easier to use than needing to know the database password for the snapshot.

I tested this out and immediately hit this because the database snapshot I used did not use the default password.

Reply:
Even after what we discussed, this is still going to make it so it only ever works with the default password since we've just made it impossible to specify what the any password other than the default. I think if we are saying "DBPassword isn't allowed to be set on the rds" that is fine but doesn't a user still need to pass `--db-password` to let us know what the snapshot password is?